x/ibc/04-channel/keeper/handshake.go

```diff
@@ -130,7 +130,7 @@ func (k Keeper) ChanOpenTry(
 		counterpartyHops, channel.Version,
 	)
 
-	consensusState, found := k.clientKeeper.GetConsensusState(
+	consensusState, found := k.clientKeeper.GetClientConsensusState(
```

Comment:
NOTE: not entirely sure if these calls should be to get the self consensus state or the client one

Reply:
cc @cwgoes

Reply:
I am actually confused as to why these functions are in this file at all, this does not match the spec. Which client is being used should be fixed & verified by the connection handshake, which has already happened. `connection.verifyChannelState` should not require a `consensusState` parameter, nor should any of the other functions in this module. See [the spec](https://github.com/cosmos/ics/tree/master/spec/ics-004-channel-and-packet-semantics#opening-handshake).

Is the intent to pass a `ConsensusState` in the message and authenticate it against a locally stored hash or something like that? If so, it should instead be a field in the message. We should never be calling client keepers directly in ICS 4; the client keeper shouldn't even be available to this module except for checking the latest height (which is just a sanity check & could be safely omitted).

Reply:
this is because ICS07 doesn't have access to state rn. Blocked by #5502. Do you suggest to call the consensus state downstream instead?

Reply:
updated